src/executables/util_sa.c

```diff
@@ -2918,15 +2928,26 @@ synccoll_check (const char *db_name, int *db_obs_coll_cnt, int *new_sys_coll_cnt
       if (check_atts)
 	{
 	  /* first drop foreign keys on attributes using collation; do not include partition sub-classes */
-	  /* CLASS_NAME, INDEX_NAME */
 
+	  // *INDENT-OFF*
 	  sprintf (query,
-		   "SELECT A.class_of.class_name, I.index_name " "from _db_attribute A, _db_index I, "
-		   "_db_index_key IK, _db_domain D " "where D.object_of = A AND D.collation_id = %d AND "
-		   "NOT (A.class_of.class_name IN (SELECT " "P.class_of.class_name " "FROM _db_partition P WHERE "
-		   "P.class_of.class_name = " "A.class_of.class_name AND P.pname IS NOT NULL))"
-		   "AND A.attr_name = IK.key_attr_name AND IK in I.key_attrs "
-		   "AND I.is_foreign_key = 1 AND I.class_of = A.class_of", db_coll->coll_id);
+		"SELECT "
+		  "[a].[class_of].[unique_name] AS [class_name], "
+		  "[i].[index_name] AS [index_name] "
+		"FROM "
+		  "[_db_attribute] AS [a] "
+		  "INNER JOIN [_db_domain] AS [d] ON [d].[object_of] = [a] "
+		  "INNER JOIN [_db_index] AS [i] ON [i].[class_of] = [a].[class_of]"
+		  "LEFT OUTER JOIN [_db_partition] AS [p] ON [p].[class_of] = [a].[class_of] "
```

Comment:
```suggestion
		  "LEFT OUTER JOIN [_db_partition] AS [p] ON [p].[class_of] = [a].[class_of] AND [p].[pname] IS NOT NULL"
```

Reply:
This change will change the result.

Reply:
Thank you.
But I thought of a way to remove the join for [_db_partition].

Reply:
Thank you. I will change it.